src/components/vm/disks/mediaEject.jsx

```diff
@@ -0,0 +1,121 @@
+import cockpit from 'cockpit';
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import { useDialogs } from 'dialogs.jsx';
+
+const _ = cockpit.gettext;
+
+export const MediaEjectModal = ({ idPrefix, vm, disk }) => {
+    const [dialogErrorDetail, setDialogErrorDetail] = useState(undefined);
+    const [inProgress, setInProgress] = useState(false);
+    const [inProgressForce, setInProgressForce] = useState(false);
+    const [defaultEjectionFailed, setDefaultEjectionFailed] = useState(false);
+
+    const Dialogs = useDialogs();
+
+    const onDelete = () => {
+        const params = {
+            connectionName: vm.connectionName,
+            id: vm.id,
+            target: disk.target,
+            eject: true,
+            live: vm.state === 'running',
+            persistent: vm.persistent,
+            force: defaultEjectionFailed
+        };
+        if (disk.type === "file") {
+            params.file = disk.source.file;
+        } else if (disk.type === "volume") {
+            params.pool = disk.source.pool;
+            params.volume = disk.source.volume;
+        } else {
+            setDialogErrorDetail(`Disk ejection is not supported for ${disk.type} disks`);
+            return;
+        }
+
+        if (!defaultEjectionFailed) {
+            setInProgress(true);
+        } else {
+            setInProgressForce(true);
+            setDialogErrorDetail(undefined);
+        }
+
+        return domainEjectDisk(params)
+                .then(() => domainGet({ connectionName: vm.connectionName, id:vm.id }))
+                .then(Dialogs.close)
+                .catch(exc => {
+                    setInProgress(false);
+                    setInProgressForce(false);
+                    setDefaultEjectionFailed(true);
+                    setDialogErrorDetail(exc.message);
+                });
+    };
+
+    let description;
+    if (disk.type === "file") {
+        description = <DescriptionListDescription id={`${idPrefix}-modal-description-file`} className="ct-monospace">{disk.source.file}</DescriptionListDescription>;
+    } else if (disk.type === "volume") {
+        description = <>
+            <DescriptionListGroup id={`${idPrefix}-modal-description-pool`} key={disk.source.pool}>
+                <DescriptionListTerm>{_("Pool")}</DescriptionListTerm>
+                <DescriptionListDescription>{disk.source.pool}</DescriptionListDescription>
+            </DescriptionListGroup>
+            <DescriptionListGroup id={`${idPrefix}-modal-description-volume`} key={disk.source.volume}>
+                <DescriptionListTerm>{_("Volume")}</DescriptionListTerm>
+                <DescriptionListDescription>{disk.source.volume}</DescriptionListDescription>
+            </DescriptionListGroup>
+        </>;
+    }
+
+    return (
+        <Modal position="top" variant="small" isOpen onClose={Dialogs.close}
+               id={idPrefix + "-modal"}
+               title={_("Eject disc from VM?")}
```

Comment:
You use `disc` with `c` at 3 places and everywhere else `disk` with `k`. I guess lets just use `disk` everywhere

Reply:
For user-visible text, the difference between disk and disc (specifically related to computer jargon):

- disk: read-write storage device with spinning platters
- disc: circular optical read-only media

There is a difference.

It's important to use the correct one in the right place, even if they're virtual. (Usually: Disc == ISO, Disk == .img/.qcow2/etc.)

Reply:
This heading comes from @garrett design suggestions: https://github.com/cockpit-project/cockpit-machines/pull/822#pullrequestreview-1131096081
Futhermore, we already use the word "disc" specifically for CD/DVD at Add disk dialog: 
![Screenshot from 2022-10-19 13-30-06](https://user-images.githubusercontent.com/42733240/196679914-62687927-5c42-4dc7-9a8d-76d63319f8f8.png)

And lastly, it seems there is gramatical difference between the word disk and disc: http://www.differencebetween.net/technology/difference-between-disc-and-disk/